module-bsp/board/rt1051/bsp/cellular/rt1051_cellular.cpp

```diff
@@ -221,15 +221,20 @@ namespace bsp
             assert(false);
         }
 
-        constexpr auto maxTXCheckRetires = 5;
+        constexpr auto maxTXCheckRetires = 10;
+        int retries;
         if (LPUART_IsEDMATxBusy(&uartDmaHandle)) {
-            for (int i = 0; i < maxTXCheckRetires; ++i) {
-                LOG_INFO("Waiting for TX free %d", i);
+            for (retries = 0; retries < maxTXCheckRetires; ++retries) {
+                LOG_INFO("Waiting for TX free %d", retries);
```

Comment:
but... is this a real fix or just a workaround? It seems that giving it more retires doesn't fix the source of the issue...

Reply:
this pr is not related to retries count rather for exit on failure

Reply:
@lucckb How is it related to exit on failure? It is not stated in issue and no other isssue is linked. Could you please elaborate.